### PR TITLE
Port #51684 to master

### PR DESCRIPTION
**Pull request by terminalmage**
See https://github.com/saltstack/salt/pull/51684

### CODE REVIEW

**Review by xeacott — DISMISSED** [review]

**Comment by Ch3LL** [comment]
@terminalmage mind fixing the merge conflict here?

File: salt/modules/boto_dynamodb.py [hunk 1]
```diff
@@ -79,12 +85,132 @@ def __virtual__():
...
+):
+    """
+    Returns a dictionary of all tags currently attached to a given resource.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt myminion boto_dynamodb.list_tags_of_resource \
+              resource_arn=arn:aws:dynamodb:us-east-1:012345678901:table/my-table
```
Comment (Ch3LL) [inline salt/modules/boto_dynamodb.py:105]: can you add a versionadded for 3006

File: salt/modules/boto_dynamodb.py [hunk 2]
```diff
@@ -79,12 +85,132 @@ def __virtual__():
...
+    """
+    Sets given tags (provided as list or dict) on the given resource.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt myminion boto_dynamodb.tag_resource \
+              resource_arn=arn:aws:dynamodb:us-east-1:012345678901:table/my-table \
+              tags='{Name: my-table, Owner: Ops}'
```
Comment (Ch3LL) [inline salt/modules/boto_dynamodb.py:148]: version added here as well

File: salt/modules/boto_dynamodb.py [hunk 3]
```diff
@@ -79,12 +85,132 @@ def __virtual__():
...
+        except botocore.exceptions.ClientError as err:
+            if retries and err.response.get("Error", {}).get("Code") == "Throttling":
+                retries -= 1
+                log.debug("Throttled by AWS API, retrying in %s seconds...", sleep)
+                time.sleep(sleep)
+                continue
+            log.error(
+                "Failed to list tags for resource %s: %s", resource_arn, err.message
+            )
+            return None
```
Comment (Ch3LL) [inline salt/modules/boto_dynamodb.py:135]: I think it makes more sense to return False here to keep in line with the other ffunctions.

File: salt/modules/boto_dynamodb.py [hunk 4]
```diff
@@ -79,12 +85,132 @@ def __virtual__():
...
+    """
+    Removes given tags (provided as list) from the given resource.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt myminion boto_dynamodb.untag_resource \
+              resource_arn=arn:aws:dynamodb:us-east-1:012345678901:table/my-table \
+              tag_keys='[Name, Owner]'
```
Comment (Ch3LL) [inline salt/modules/boto_dynamodb.py:188]: versionadded here as well

**Review by Ch3LL — CHANGES_REQUESTED** [review]
This will also require a changelog

**Comment by terminalmage** [comment]
@Ch3LL I thought I already had rebased onto master, but I might be mistaken. I did a rebase, and I think I've covered all the review comments.

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**